Transform: added merge transform that will merge multiple series/tables into one table (#25490)
grafana/grafana: docs/sources/panels/transformations.md

{{< docs-imagebox img="/img/docs/transformations/reduce-after-7-0.png" class="docs-image--no-shadow" max-width= "1100px" >}}

### Merge

> Use this transformation to _merge_ the result from multiple queries into one single result based on the time field. This is really helpful when using the table panel visualization. 

Reviewer: ```suggestion
Use this transformation to combine the result from multiple queries into one single result based on the time field. This is helpful when using the table panel visualization. 
```

In the example below, we are visualizing multiple queries returning table data before applying the transformation.

{{< docs-imagebox img="/img/docs/transformations/table-data-before-merge-7-1.png" class="docs-image--no-shadow" max-width= "1100px" >}}

> Below you can see the same example after applying the merge transformation.

Reviewer: ```suggestion
Here is the same example after applying the merge transformation.
```

{{< docs-imagebox img="/img/docs/transformations/table-data-after-merge-7-1.png" class="docs-image--no-shadow" max-width= "1100px" >}}

> If any of the queries return time series data a `Metric` column containing the name of the query will be added. This value can be customized by defining `Label` on the source query.

Reviewer: ```suggestion
If any of the queries return time series data, then a `Metric` column containing the name of the query is added. You can customize this value by defining `Label` on the source query.
```

> If any the queries returns time series data a `Metric` column containing the name of the query will be added. This value can be customized e.g. by defining `Label` on the source query.

Reviewer 2: ```suggestion
If any of the queries return time series data a `Metric` column containing the name of the query will be added. This value can be customized e.g. by defining `Label` on the source query.
```

In the example below, we are visualizing multiple queries returning time series data before applying the transformation.

{{< docs-imagebox img="/img/docs/transformations/time-series-before-merge-7-1.png" class="docs-image--no-shadow" max-width= "1100px" >}}

> Below you can see the same example after applying the merge transformation.

Reviewer: ```suggestion
Here is the same example after applying the merge transformation.
```
